libs/wallet/src/api/state/multiInjectedProvidersAtom.ts

```diff
@@ -0,0 +1,37 @@
+import { atom } from 'jotai'
+import { atomWithStorage } from 'jotai/utils'
+
+import { getJotaiIsolatedStorage, jotaiStore } from '@cowprotocol/core'
+
+import { EIP6963AnnounceProviderEvent, EIP6963ProviderDetail } from '../eip6963-types'
+
+export const multiInjectedProvidersAtom = atom<EIP6963ProviderDetail[]>([])
+
+// RDNS of the selected EIP-6963 provider
+export const selectedEip6963ProviderRdnsAtom = atomWithStorage<string | null>(
+  'selectedEip6963ProviderAtom:v1',
```

Comment:
Is this the first? Start with 0?

```suggestion
  'selectedEip6963ProviderAtom:v0,
```

Reply:
Fixed https://github.com/cowprotocol/cowswap/pull/4420/commits/c98d027b744ea771203d50a39eda74cb1f7a0215